sklearn/metrics/classification.py

```diff
@@ -1191,6 +1191,29 @@ def classification_report(y_true, y_pred, labels=None, target_names=None,
     target_names : list of strings
         Optional display names matching the labels (same order).
 
+    average : None, string or tuple of strings, which are a subset of [
+        'weighted', 'micro', 'macro', 'samples']
+        This determines the averaging methods shown in the report:
+
+        ``'micro'``:
+            Calculate metrics globally by counting the total true positives,
+            false negatives and false positives.
+        ``'macro'``:
+            Calculate metrics for each label, and find their unweighted
+            mean.  This does not take label imbalance into account.
+        ``'weighted'``:
+            Calculate metrics for each label, and find their average, weighted
+            by support (the number of true instances for each label). This
+            alters 'macro' to account for label imbalance; it can result in an
+            F-score that is not between precision and recall.
+        ``'samples'``:
+            Calculate metrics for each instance, and find their average (only
+            accepted for multilabel classification where this differs from
+            :func:`accuracy_score`).
+
+        if average = None (default), results for all relevant averaging
+        methods are shown
```

Comment:
I wouldn't add the `sample` averaging by default as it is only for multilabel classification.